src/bootstrap/bin/rustc.rs

```diff
@@ -305,6 +305,16 @@ fn main() {
                 cmd.arg("-C").arg("target-feature=-crt-static");
             }
         }
+
+        let crate_type = args.windows(2)
+            .find(|w| &*w[0] == "--crate-type")
+            .and_then(|w| w[1].to_str());
+
+        if let Some("proc-macro") = crate_type {
+            if let Ok(map) = env::var("RUSTC_DEBUGINFO_MAP") {
+                cmd.arg("--remap-path-prefix").arg(&map);
+            }
+        }
```

Comment:
Instead of doing this just for proc macro crates, can we just universally pass the remap to all invocations of rustc? We already do so earlier in this file if `target.is_some()`, I'd like to not duplicate that.

Reply:
That would certainly be fine with me.  I did it this way to apply the change to as few crates as possible to be safe, but if it's correct that is indeed cleaner.

Reply:
To be clear, I don't know why we're not already doing that (maybe just by accident).

Reply:
It looks like doing it this way does help make other crates reproducible (e.g., plugins like rust-clippy), so I'll do it that way.